packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Providers/MrwSerializationTypeProvider.cs

```diff
@@ -31,12 +31,12 @@ internal sealed class MrwSerializationTypeProvider : TypeProvider
         private readonly CSharpType? _iJsonModelObjectInterface;
         private readonly CSharpType _iPersistableModelTInterface;
         private readonly CSharpType? _iPersistableModelObjectInterface;
-        private ModelProvider _model;
+        private TypeProvider _model;
         private InputModelType _inputModel;
         private readonly FieldProvider? _rawDataField;
         private readonly bool _isStruct;
 
-        public MrwSerializationTypeProvider(ModelProvider model, InputModelType inputModel)
+        public MrwSerializationTypeProvider(TypeProvider model, InputModelType inputModel)
```

Comment:
Shouldn't this one still be ModelProvider? Especially since the second parameter is InputModelType.

Reply:
I don't think we want to force customers to inherit from our ModelProvider if they want to completely write a new type of model so TypeProvider allows them to do this.

Reply:
does this really matter because this class is internal, no one could override this.

Reply:
The calling signature needs to change to TypeProvider so this also needs to change.  The fact that nothing has broken is proof that ModelProvider is not special here and MrwSerializationProvider can work with any TypeProvider.